### PR TITLE
support saving audio track and subtitle tracks in history

### PR DESCRIPTION
**Pull request by adipose**
This seems to work pretty well for videos and DVDs.  It seems pretty simple, but maybe there are some scenarios I haven't thought of.  Worth testing, but for 4-5 videos I've tested, it works.

### CODE REVIEW

File: src/mpc-hc/AppSettings.cpp [hunk 1]
```diff
@@ -2860,6 +2860,42 @@ DVD_POSITION CAppSettings::CRecentFileListWithMoreInfo::GetCurrentDVDPosition()
     return DVD_POSITION();
 }
 
+void CAppSettings::CRecentFileListWithMoreInfo::UpdateCurrentAudioTrack(int audioIndex) {
+    size_t idx;
+    if (GetCurrentIndex(idx)) {
+        if (rfe_array[idx].AudioTrackIndex != audioIndex) {
+            rfe_array[idx].AudioTrackIndex = audioIndex;
+            WriteMediaHistoryEntry(rfe_array[idx]);
```
Comment (clsid2) [inline src/mpc-hc/AppSettings.cpp]: Maybe add function that only updates the audioindex/subindex value? Similar to how abRepeat has such as function. That saves a lot of write actions, specially when toggling through available tracks.

Comment (adipose) [reply src/mpc-hc/AppSettings.cpp]: Ah good idea, I see what you mean.

File: src/mpc-hc/MainFrm.cpp [hunk 2]
```diff
@@ -14667,13 +14686,24 @@ bool CMainFrame::OpenMediaPrivate(CAutoPtr<OpenMediaData> pOMD)
         OpenSetupWindowTitle();
         checkAborted();
 
-        int audstm = SetupAudioStreams();
+        int audstm;
+
+        if (m_loadedAudioTrackIndex != -1) {
```
Comment (clsid2) [inline src/mpc-hc/MainFrm.cpp]: Test what happens if the stored index is larger than the number of available tracks.

File: src/mpc-hc/AppSettings.cpp [hunk 3]
```diff
@@ -3100,6 +3136,9 @@ bool CAppSettings::CRecentFileListWithMoreInfo::LoadMediaHistoryEntry(CStringW h
         }
         r.subs.AddTail(st);
     }
+
+    r.AudioTrackIndex = pApp->GetProfileIntW(subSection, L"AudioTrackIndex", -1);
```
Comment (clsid2) [inline src/mpc-hc/AppSettings.cpp:3140]: maybe add sanity check for invalid values
`if (value < -1) value = -1`

File: src/mpc-hc/MainFrm.cpp (hunk 2 above)
Comment (adipose) [reply src/mpc-hc/MainFrm.cpp]: what happens is that `pSS->Enable(sidx, AMSTREAMSELECTENABLE_ENABLE)` fails.  The code doesn't check the return value normally.

**Comment by adipose** [comment]
Maybe overkill on some checks, not sure.

Also I noticed that this code doesn't check return:

```
            if (!(dwFlags & AMSTREAMSELECTINFO_EXCLUSIVE)) {
                pSSF->Enable(i, AMSTREAMSELECTENABLE_ENABLE);
            }
```
in `SetSubtitle()`

Code is a bit wonky so I'm not sure if returning failure at that point is a good idea or not.

**Pull request merged**